drivers: wifi: AIROC: add support of new devices

npal-cy opened the pull request:
- add support Wi-Fi Host Driver (WHD) 4.2.1
- add Country configuration from Kconfig

Support new devices:
 -- CYW43022
 -- CYW55500 (WIFI6)
 -- CYW55572 (WIFI6)

Support new modules:
 -- CYW43022CUB
 -- CYW955573M2IPA1_SM
 -- CYW955513SDM2WLIPA_SM
 -- CYW4373_MURATA_2BC
 -- CYW4373_MURATA_2AE
 -- CYW43439_STERLING_LWBPLUS

npal-cy committed:
drivers: wifi: AIROC: add support of new devices

- add support Wi-Fi Host Driver (WHD) 4.2.1
- add Country configuration from Kconfig

Support new devices:
 -- CYW43022
 -- CYW55500 (WIFI6)
 -- CYW55572 (WIFI6)

Support new modules:
 -- CYW43022CUB
 -- CYW955573M2IPA1_SM
 -- CYW955513SDM2WLIPA_SM
 -- CYW4373_MURATA_2BC
 -- CYW4373_MURATA_2AE
 -- CYW43439_STERLING_LWBPLUS

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

jukkar commented on drivers/wifi/infineon/airoc_whd_hal_common.c:
```diff
@@ -64,21 +64,47 @@ int airoc_wifi_power_on(const struct device *dev)
 
+inline void whd_mem_memcpy(void *dest, const void *src, size_t len)
+{
+#ifdef PROTO_MSGBUF
```
What is the idea of the `PROTO_MSGBUF`? I could not find it being defined anywhere in the code.

npal-cy committed:
drivers: wifi: AIROC: update connect\scan api

- Updated airoc_mgmt_connect to be able use security provided
  by user (params->security). If it is not passed, we will perform
  scan ssid to detect secure type.

- Fix convert_whd_security_to_zephyr() with correct corresponding
  whd_security_t to zephyr security type.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented:
Hi @jukkar i updated west.yaml with updated hal_infoneon (not sure why DNM still here). This PR ready for merge.

ifyall commented:
@Deepak-AlifSemi, this patch adds support for the CYW955513SDM2WLIPA, which is a package of the 55513 series. Because the 2FY datasheet isn't yet released, I can't comment on if this is the package used there. So, it may work if you use the CYW955513SDM2WLIPA_SM KConfig value in your application.

I can say that our Wireless team is working on bringing a more fully qualified PR with support for CYW55513 and other devices.

I have sent this issue link to one of the Wireless team members and will ask him to comment.

Ian

kartben commented:
> I was thinking that merge windows is closed and only fixes should be present. Can someone explain this ?
> 
> CC: @kartben

mmm what do you mean?

<img width="603" alt="image" src="https://github.com/user-attachments/assets/ea160ee0-41f0-40e6-9e15-dea92195d78e" />

nandojve commented:
Uhmm,

I saw in GH list this was "merged" and then when I came here I saw the message about PR was merged. So, each time we comment here that message shows up.

![image](https://github.com/user-attachments/assets/a1628648-9065-4b12-b5dc-66df9b074ef4)

I got confused.
Sorry about the noise.
